libraries/Microsoft.Bot.Builder.Ai/LocaleConverter.cs

```diff
@@ -11,41 +11,63 @@
 
 namespace Microsoft.Bot.Builder.Ai
 {
-
-    //Struct used to store date format and time format for different locales
+    /// <summary>
+    /// DateAndTimeLocaleFormat Class used to store date format and time format
+    /// for different locales.
+    /// </summary> 
     internal class DateAndTimeLocaleFormat
-    { 
+    {
         public string TimeFormat { get; set; }
-        public string DateFormat { get;set; }
+        public string DateFormat { get; set; }
 
     }
 
-    //Struct to store  text and date time object from Microsoft Recognizer recognition result
+    /// <summary>
+    /// TextAndDateTime Class used to store  text and date time object 
+    /// from Microsoft Recognizer recognition result.
+    /// </summary> 
     internal class TextAndDateTime
-    { 
+    {
         public string Text { get; set; }
-        public DateTime DateTimeObject { get; set; }
+        public DateTime dateTime { get; set; }
     }
 
     /// <summary>
-    /// Locale Converter Class used to convert between locales
-    /// in terms of date and time only.
+    /// Locale Converter Class Converts dates and times 
+    /// between different locales.
     /// </summary>
     public class LocaleConverter : ILocaleConverter
-    { 
-        private Dictionary<string, DateAndTimeLocaleFormat> mapLocaleToFunction = new Dictionary<string, DateAndTimeLocaleFormat>();
+    {
 
-        public LocaleConverter()
+        private static readonly Dictionary<string, DateAndTimeLocaleFormat> _mapLocaleToFunction = new Dictionary<string, DateAndTimeLocaleFormat>();
+        private static object _lockMap = new object();
+        private static LocaleConverter _localeConverter;
+        private static readonly Dictionary<string, DateTimeModel> _cacheDateTimeModel = new Dictionary<string, DateTimeModel>();
+        public static LocaleConverter Converter
+        {
+            get
+            {
+
+                if (_localeConverter == null)
+                {
+                    _localeConverter = new LocaleConverter();
+                }
+                return _localeConverter;
+            }
+        }
+        private LocaleConverter()
         {
-            initLocales();
+            InitLocales();
         }
-        
+
         /// <summary>
         /// Init different locales format,
         /// Supporting English, French, Deutsche and Chinese Locales.
         /// </summary>
-        private void initLocales()
+        private void InitLocales()
         {
+            if (_mapLocaleToFunction.Count > 0 && _cacheDateTimeModel.Count > 0)
```

Comment:
Bug: Now that the dictionaries are static, you can't call these methods w/o putting them inside a lock. Or switching to a concurrent dictionary.